proto/astarteplatform/msghub/config.proto

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+syntax = "proto3";
+
+package astarteplatform.msghub;
+
+message ConfigMessage {
+  string realm = 1;
+  string device_id = 2;
+  optional string credentials_secret = 3;
+  string pairing_url = 4;
+  optional string pairing_token = 5;
+}
+
+message ConfigResponse {
+  bool success = 1;
+}
+
+service AstarteMessageHubConfig {
+  rpc SetConfig (ConfigMessage) returns (ConfigResponse);
```

Comment:
In this case, you can use as return type `google.protobuf.Empty`, it is enough to describe the success or failure of `SetConfig` method without define another message.
``` proto
import "google/protobuf/empty.proto";

...

rpc SetConfig (ConfigMessage)  returns (google.protobuf.Empty){}
```
In case of error you can send a gRPC status describing the result of an RPC call.